packages/mcdoc/src/runtime/checker/error.ts

```diff
@@ -0,0 +1,463 @@
+	| RangeError<T>
+	| TypeMismatchError<T>
+	| MissingKeyError<T>
+
+export interface McdocRuntimeBaseError<T> {
+	node: RuntimeNode<T>
+	/**
+	 * This is set when this error may not need to be fixed if another error of the same kind is
+	 * fixed instead. This contains a list of nodes with conflicting
+	 */
+	nodesWithConflictingErrors?: RuntimeNode<T>[]
+}
+
+export interface SimpleError<T> extends McdocRuntimeBaseError<T> {
+	kind:
+		| 'duplicate_key'
+		| 'unknown_key'
+		| 'expected_key_value_pair'
+		| 'unknown_tuple_element'
+		| 'internal'
+}
+export namespace SimpleError {
+	export function is<T>(error: McdocRuntimeError<T> | undefined): error is SimpleError<T> {
+		return error?.kind === 'duplicate_key'
+			|| error?.kind === 'unknown_key'
+			|| error?.kind === 'expected_key_value_pair'
+			|| error?.kind === 'unknown_tuple_element'
+			|| error?.kind === 'internal'
+	}
+}
+
+export interface UnknownKeyError<T> extends McdocRuntimeBaseError<T> {
+	kind: 'unknown_key'
+}
+export namespace UnknownKeyError {
+	export function is<T>(
+		error: McdocRuntimeError<T> | undefined,
+	): error is UnknownKeyError<T> {
+		return error?.kind === 'unknown_key'
+	}
+}
+
+export interface RangeError<T> extends McdocRuntimeBaseError<T> {
+	kind:
+		| 'invalid_collection_length'
+		| 'invalid_string_length'
+		| 'number_out_of_range'
+	ranges: NumericRange[]
+}
+export namespace RangeError {
+	export function is<T>(error: McdocRuntimeError<T> | undefined): error is RangeError<T> {
+		return error?.kind === 'invalid_collection_length'
+			|| error?.kind === 'invalid_string_length'
+			|| error?.kind === 'number_out_of_range'
+	}
+}
+
+export interface MissingKeyError<T> extends McdocRuntimeBaseError<T> {
+	kind: 'missing_key'
+	/**
+	 * A list of multiple are an alternative, and at least on e needs to be fixed, not neccessarily
+	 * all of them.
+	 *
+	 * In case there are multiple non-conflicting missing keys, multiple errors will be reported on
+	 * the same node instead.
+	 *
+	 * If this has a length > 1, there is at least one error that needs to be fixed, and at least one
+	 * error that does not neccassarily need to be fixed.
+	 */
+	keys: string[]
+}
+export namespace MissingKeyError {
+	export function is<T>(error: McdocRuntimeError<T> | undefined): error is MissingKeyError<T> {
+		return error?.kind === 'missing_key'
+	}
+}
+
+export interface TypeMismatchError<T> extends McdocRuntimeBaseError<T> {
+	kind: 'type_mismatch'
+	expected: SimplifiedMcdocTypeNoUnion[]
+}
+export namespace TypeMismatchError {
+	export function is<T>(error: McdocRuntimeError<T> | undefined): error is TypeMismatchError<T> {
+		return error?.kind === 'type_mismatch'
+	}
+}
+
+export interface ErrorCondensingDefinition<T> {
+	definition: CheckerTreeDefinitionNode<T>
+	errors: McdocRuntimeError<T>[]
+}
+
+export function condenseErrorsAndFilterSiblings<T>(
+	definitions: {
+		definition: CheckerTreeDefinitionNode<T>
+		errors: McdocRuntimeError<T>[]
+	}[],
```

Comment:
This should be `definitions: ErrorCondensingDefinition<T>[]`

Reply:
Yeah I think this was nuked by a merge conflict

Reply:
Ah yeah good catch, this broke when I copied stuff over again after merging `internal` error. :upside_down_face: